### DISCUSSION TITLE
Android: File Manager Access For Android.

### DISCUSSION THREAD

**Discussion by llama-nl** [godotengine/godot-proposals#10462] [Platforms]
Actually I am working on a project which needs to access file. I know Godot has file access node. But like other engine Godot doesn't have comfortable way to access file on Android.

**Comment by AThousandShips** [comment]
What do you mean by "file access node"? And "comfortable way to access file"? What do you need that isn't covered by [`FileDialog`](https://docs.godotengine.org/en/stable/classes/class_filedialog.html)?

**Reply by llama-nl** [reply]
Sorry, for confusion.
Yes I am talking about file **FileDialog**. But for Android device. 

**FileDialog** node is good for Desktop UI but it is not looking good for Android. Sometime it annoys a lot. 

I hope you'll understand.

**Reply by MarkoGrbec** [reply]
Do you mean annoys -> can't access files when you are in wrong section of directory?

**Reply by llama-nl** [reply]
No, Let me explain. 

Actually I was talking about OS-specific file dialogs.
See FileDialog, it looks good for desktop. But when you'll use it on mobile platform do you think anyone will like to access files like desktop?

Here is a example video:
https://github.com/user-attachments/assets/217480e3-5250-4c6a-a5e2-a9e10f8d09b9

It is a Unity3D application [Prisma3d](https://play.google.com/store/apps/details?id=com.prisma3D.prisma3D)
 When I am clinking on **3D Import*/3D Export* it is opening my android File Manager. Not FileDialog like Godot.

**Comment by Calinou** [comment]
Proposal issue opened:

- https://github.com/godotengine/godot-proposals/issues/10983
